neqo-transport/src/connection/mod.rs

```diff
@@ -2591,39 +2431,21 @@ impl Connection {
                 qdebug!([self], "Lost: {:?}", token);
                 match token {
                     RecoveryToken::Ack(_) => {}
-                    RecoveryToken::Stream(st) => self.send_streams.lost(&st),
                     RecoveryToken::Crypto(ct) => self.crypto.lost(&ct),
-                    RecoveryToken::Flow(ft) => {
-                        self.flow_mgr.borrow_mut().lost(&ft, &mut self.indexes)
-                    }
                     RecoveryToken::HandshakeDone => self.state_signaling.handshake_done(),
                     RecoveryToken::NewToken(seqno) => self.new_token.lost(*seqno),
                     RecoveryToken::NewConnectionId(ncid) => self.cid_manager.lost(ncid),
                     RecoveryToken::RetireConnectionId(seqno) => self.paths.lost_retire_cid(*seqno),
-                    RecoveryToken::ResetStream { stream_id } => {
-                        self.send_streams.reset_lost(*stream_id)
-                    }
-                    RecoveryToken::DataBlocked(limit) => {
-                        self.flow_control_sender.borrow_mut().lost(*limit)
-                    }
-                    RecoveryToken::StreamDataBlocked { stream_id, limit } => {
-                        self.send_streams.blocked_lost(*stream_id, *limit)
-                    }
-                    RecoveryToken::MaxData(maximum_data) => {
-                        self.flow_control_receiver.borrow_mut().lost(*maximum_data)
-                    }
-                    RecoveryToken::MaxStreamData {
-                        stream_id,
-                        max_data,
-                    } => {
-                        if let Ok((_, Some(rs))) = self.obtain_stream(*stream_id) {
-                            rs.max_stream_data_lost(*max_data);
-                        }
-                    }
-                    RecoveryToken::StopSending { stream_id } => {
-                        if let Ok((_, Some(rs))) = self.obtain_stream(*stream_id) {
-                            rs.stop_sending_lost();
-                        }
+                    RecoveryToken::Stream(_)
+                    | RecoveryToken::ResetStream { .. }
+                    | RecoveryToken::StreamDataBlocked { .. }
+                    | RecoveryToken::MaxStreamData { .. }
+                    | RecoveryToken::StopSending { .. }
+                    | RecoveryToken::StreamsBlocked { .. }
+                    | RecoveryToken::MaxStreams { .. }
+                    | RecoveryToken::DataBlocked(_)
+                    | RecoveryToken::MaxData(_) => {
+                        self.streams.lost(token);
```

Comment:
I mention this later, but maybe RecoveryToken::stream_related() is worth defining so that this match block doesn't appear in multiple places.